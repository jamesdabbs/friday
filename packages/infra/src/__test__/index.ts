import { Testing } from 'cdk8s';

export function synth(construct: any, { name = 'construct', ...params }: Record<string, unknown> = {}) {
  const app = Testing.app();
  const chart = new construct(app, name, params);
  return Testing.synth(chart);
}

export function find(chart: any[], kind: string, name?: string) {
  return chart.find(value => {
    if (value.kind !== kind) { return false }
    if (name && value.metadata?.name !== name) { return false }

    return true
  })
}
