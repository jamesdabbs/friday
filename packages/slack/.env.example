SLACK_BOT_USER_TOKEN=...
SLACK_SIGNING_SECRET=...
