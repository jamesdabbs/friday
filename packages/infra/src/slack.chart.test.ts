import { SlackChart } from './slack.chart';
import { synth } from './__test__';

describe(SlackChart, () => {
  describe('with defaults', () => {
    const chart = synth(SlackChart)

    it('matches the snapshot', () => {
      expect(chart).toMatchSnapshot()
    })
  })
});
