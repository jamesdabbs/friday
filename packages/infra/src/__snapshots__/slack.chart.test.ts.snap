// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlackChart with defaults matches the snapshot 1`] = `
Array [
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "name": "slack",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "cdk8s.deployment": "construct-slack.deployment-c8b9c8f1",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "cdk8s.deployment": "construct-slack.deployment-c8b9c8f1",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "env": Array [
                Object {
                  "name": "SLACK_SIGNING_SECRET",
                  "valueFrom": Object {
                    "secretKeyRef": Object {
                      "key": "SLACK_SIGNING_SECRET",
                      "name": "slack-secret",
                    },
                  },
                },
                Object {
                  "name": "SLACK_BOT_USER_TOKEN",
                  "valueFrom": Object {
                    "secretKeyRef": Object {
                      "key": "SLACK_BOT_USER_TOKEN",
                      "name": "slack-secret",
                    },
                  },
                },
              ],
              "image": "slack",
              "imagePullPolicy": "IfNotPresent",
              "name": "slack",
              "ports": Array [],
              "volumeMounts": Array [],
            },
          ],
          "volumes": Array [],
        },
      },
    },
  },
]
`;
